src/relay/transforms/simplify_expr.cc

```diff
@@ -585,6 +585,47 @@ class EliminateIdentityRewrite : public DFPatternRewrite {
   DFPattern const_;
 };
 
+/*! \brief Make two consecutive add able to be constant_folded. */
+class SimplifyConsecutiveAdd : public DFPatternRewrite {
+ public:
+  SimplifyConsecutiveAdd() {
+    x_ = IsWildcard();
+    const1_ = IsConstant();
+    const2_ = IsConstant();
+    DFPattern add_op1_ = IsOp("add");
+    DFPattern add_op2_ = IsOp("add");
+    pattern_ = add_op2_({add_op1_({x_, const1_}), const2_});
```

Comment:
1. Only private class members need the underline suffix.
2. You actually don't need two `add_op`s because all ops are the same.
3. Might be better to add a comment saying pattern matching supports commutative property for addition so this pattern already covers all possible cases.
```suggestion
    DFPattern add_op = IsOp("add");
    pattern_ = add_op({add_op({x_, const1_}), const2_});
```